File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda box: box[1], reverse=True)
        res=0
        for numberOfBoxes, unitsPerBox in boxTypes:
            numBoxes=min(truckSize, numberOfBoxes)
            res+=(numBoxes*unitsPerBox)
            truckSize-=numBoxes
            
        return res

File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.cpp
class Solution {
public:
    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
        priority_queue<int> pq;
        int res=0;
        for(int i=0; i<boxTypes.size(); i++){
            for(int j=0; j<boxTypes[i][0]; j++){
                pq.push(boxTypes[i][1]);
            }
        }
        for(int i=0; i<truckSize; i++){
            if(!pq.empty()){
                res+=pq.top();
                pq.pop();
            }
        }
        return res;
    }
};